src/btree/col_modify.c

```diff
@@ -39,22 +39,29 @@ __wt_col_modify(WT_CURSOR_BTREE *cbt, uint64_t recno, const WT_ITEM *value, WT_U
     append = logged = false;
 
     /*
-     * We should have EITHER:
-     * - A full update list to instantiate with.
-     * - An update to append the existing update list with.
+     * We should have one of the following:
+     * - A full update list to instantiate.
+     * - An update to append to the existing update list.
      * - A key/value pair to create an update with and append to the update list.
+     * - A key with no value to create a reserved or tombstone update to append to the update list.
+     *
+     * A "full update list" is distinguished from "an update" by checking whether it has a "next"
+     * update.
      */
-    WT_ASSERT(session, (value == NULL && upd_arg != NULL) || (value != NULL && upd_arg == NULL));
+    WT_ASSERT(session, ((modify_type == WT_UPDATE_RESERVE || modify_type == WT_UPDATE_TOMBSTONE) &&
+                         value == NULL && upd_arg == NULL) ||
+        (modify_type != WT_UPDATE_TOMBSTONE && ((value == NULL && upd_arg != NULL) ||
```

Comment:
Could you clarify why this isn't WT_UPDATE_RESERVE too? As you check against TOMBSTONE and RESERVED in the other scenario.

Reply:
My  mistake, thanks! Fixed in 20db497bf.